4.6.0

qiyuewuyi opened the pull request.

Commit:
expose logger, retry strategy and thread pool on interface to clients

Commit:
shared Logger in clients with BatchWriter's

Commit:
update version to 4.6.0

Commit:
changes according to Zhe's comments

qiyuewuyi committed:
Merge pull request #20 from TimeExceed/expose_config

Expose config

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/aliyun/aliyun-tablestore-cpp-sdk?pullRequest=21) <br/>All committers have signed the CLA.

TimeExceed committed:
fix RangeIterator crashes (#22)

* fix RangeIterator crashes

by turning to SyncClient

* nextStart() and consumedCapacity() of RangeIterator

and more changes according to Zhe's comments.

TimeExceed committed:
remove dependency on libuuid (#23)

* getHostName

* move util::random::Random to util

* put util::Random onto Tracker::create()

* base57

* remove dependency to libuuid

Now a tracker is composed of 16-bit hostname (adler32 of hostname and fold to 16-bit) and 48-bit random bits, and then encoded by base57.

* changes according to review comments

1. tabulate base57 decoding
2. resize instead of reserve in getHostName

* changes according to review comments

TimeExceed committed:
[fix #4] use different serialization functions of TableOptions for CreateTableRequest and UpdateTableRequest (#24)
